test/e2e/01-install.sh

```diff
@@ -66,13 +68,17 @@ set -e
 kubectl create secret tls -n tekton-pipelines tekton-results-tls --cert="${SSL_CERT_PATH}/tekton-results-cert.pem" --key="${SSL_CERT_PATH}/tekton-results-key.pem" || true
 
 echo "Installing Tekton Results..."
-kubectl kustomize "${ROOT}/test/e2e/kustomize" | ko apply -f -
+if [[ "$(uname -a)" == *ARM64* && "$(uname -m)" == x86_64* ]]; then
```

Comment:
For the first match I'll suggest using Bash's regex match operator `=~` which is better suited for this type of check, so `"$(uname -a)" == *ARM64*` could become `"$(uname -a)" =~ ARM64`.

In the second part, why do you need the `*` in `x86_64*`? If `"$(uname -m)" == "x86_64"` does not match, I'll suggest again using the `=~` operator.

Instead of repeating the `kubectl kustomize ...` command in the if and else clauses, you can do:
`[[ "$(uname -a)" == *ARM64* && "$(uname -m)" == x86_64* ]] && extra_ko_params="--platform=linux/arm64"`
and then
`kubectl kustomize "${ROOT}/test/e2e/kustomize" | ko apply $extra_ko_params -f -`. If `extra_ko_params` is not set, it'll return empty string which won't affect the `ko` command.

Reply:
Done

Reply:
@enarha empty string throws error
```
error: Unexpected args: []
See 'kubectl apply -h' for help and examples
```

Reply:
The problem is you've put quotes around `$extra_ko_params`. Without them the proposed code works.

```[ results] -> kubectl kustomize "./test/e2e/kustomize" | ko apply "$extra_ko_params" -f -
error: Unexpected args: []
See 'kubectl apply -h' for help and examples```

```[ results] -> kubectl kustomize "./test/e2e/kustomize" | ko apply $extra_ko_params -f -
2023/01/30 20:38:51 Using base distroless.dev/static:latest@sha256:01dfcc8d455ddb111f496eca6e5fe88aa5efac96a7f08b269f1a259eb5a2f5fc for github.com/tektoncd/results/cmd/api
...```

Anyway, Adam's suggestion to always set the platform is a better approach.

Reply:
> The problem is you've put quotes around $extra_ko_params

ohh damn, I put quotes to avoid shellcheck error. But yes I think always putting `extra_ko_params` should work